### ISSUE TITLE
Regarding sensitivity of hyperparameters

### ISSUE THREAD

**Issue by srbhcs** [bytedance/LVLM_Interpretation#12]
Hi Authors, 

<img width="451" height="435" alt="Image" src="https://github.com/user-attachments/assets/dc0f4e38-1c6a-49fb-9403-0c1e379f15e1" />

Need your help. I have tried a lot with various hyperparameters to generate the overlay map for the above image and prompt, but have not been able to get a good map. Can you please help me see where I am going wrong?

Command: python3 main.py --method iGOS+ --model llava_next --dataset llava-bench --data_path questions_llava.jsonl --image_folder /storage --output_dir files --size 16 --L1 3.0 --L2 0.1 --L3 10.0 --ig_iter 4 --gamma 1.0 --iterations 5 --momentum 5

Output Text:

prompt: A bounding box is an annotated rectangle surrounding an object. The edges of bounding boxes should touch the outermost pixels of the object that is being labeled. Given the two bounding boxes on the image, labeled, provide the label of the bounding box which more accurately localizes and encloses the tank top (clothing).
answer: The bounding box labeled "A" is more accurately localizing and enclosing the tank top (clothing) than the other bounding box. 
keywords: labeled,A
Deletion (Avg.): 0.21198
Insertion (Avg.): 0.74728
Time (Avg.): 18.725

Logs: 

Loading checkpoint shards: 100%|██████████████████████████████████████████████████████████████| 4/4 [00:05<00:00,  1.30s/it]
Model Class: LlavaMistralForCausalLM
total number of data samples: 122
/storage/miniconda3/envs/li/lib/python3.10/site-packages/transformers/generation/configuration_utils.py:392: UserWarning: `do_sample` is set to `False`. However, `temperature` is set to `0` -- this flag is only used in sample-based generation modes. You should set `do_sample=True` or unset `temperature`.
  warnings.warn(
/storage/miniconda3/envs/li/lib/python3.10/site-packages/transformers/generation/configuration_utils.py:397: UserWarning: `do_sample` is set to `False`. However, `top_p` is set to `None` -- this flag is only used in sample-based generation modes. You should set `do_sample=True` or unset `top_p`.
  warnings.warn(
The attention mask and the pad token id were not set. As a consequence, you may observe unexpected behavior. Please pass your input's `attention_mask` to obtain reliable results.
Setting `pad_token_id` to `eos_token_id`:2 for open-end generation.
/storage/miniconda3/envs/li/lib/python3.10/site-packages/torch/_dynamo/eval_frame.py:745: UserWarning: torch.utils.checkpoint: the use_reentrant parameter should be passed explicitly. In version 2.5 we will raise an exception if use_reentrant is not passed. use_reentrant=False is recommended, but if you need to preserve the current default behavior, you can pass use_reentrant=True. Refer to docs for more details on the differences between the two variants.
  return fn(*args, **kwargs)
/storage/miniconda3/envs/li/lib/python3.10/site-packages/torch/utils/checkpoint.py:87: UserWarning: None of the inputs have requires_grad=True. Gradients will be None
  warnings.warn(
iteration: 0 lr: 10.0000 loss_del: -1.5930, loss_ins: -3.2062, loss_l1: 0.0000, loss_tv: 0.0000, loss_l2: 0.0000
iteration: 1 lr: 10.0000 loss_del: -3.2427, loss_ins: -1.5823, loss_l1: 0.8898, loss_tv: 2.6591, loss_l2: 2.2759
iteration: 2 lr: 10.0000 loss_del: -4.4422, loss_ins: -1.5746, loss_l1: 1.1457, loss_tv: 3.6587, loss_l2: 1.1567
iteration: 3 lr: 10.0000 loss_del: -7.3335, loss_ins: -1.6267, loss_l1: 1.1915, loss_tv: 4.0673, loss_l2: 0.4628
iteration: 4 lr: 10.0000 loss_del: -2.0724, loss_ins: -3.0416, loss_l1: 1.2632, loss_tv: 4.1228, loss_l2: 0.1807
 Deletion (Avg.): 0.34245 Insertion (Avg.): 0.78562 Time (Avg.): 18.691

Overlay Map: 

<img width="336" height="336" alt="Image" src="https://github.com/user-attachments/assets/fea4026b-6b2f-4917-b293-c9a30053325f" />

**Comment by srbhcs** [comment]
I have done just one modification in the main.py script by changing the model path.

    elif args.model == 'llava_next':
        model_path = "liuhaotian/llava-v1.6-mistral-7b"
        model_name = get_model_name_from_path(model_path)
        tokenizer, model, image_processor, context_len = load_pretrained_model_next(model_path, args.model_base, model_name, device_map='auto')

And adding llava_mistral.py (taken from the official LLaVA_NeXt Repository) in the llava_next path.

**Comment by srbhcs** [comment]
The logs are complete, except for the warning below.
  warnings.warn(
/storage/miniconda3/envs/li/lib/python3.10/site-packages/torch/nn/modules/module.py:2397: UserWarning: for vision_model.encoder.layers.19.self_attn.v_proj.bias: copying from a non-meta parameter in the checkpoint to a meta parameter in the current model, which is a no-op. (Did you mean to pass `assign=True` to assign items in the state dictionary to their corresponding key in the module instead of copying them in place?)

pip list:

Package                  Version      Editable project location
------------------------ ------------ ----------------------------
accelerate               1.13.0
aiohappyeyeballs         2.6.1
aiohttp                  3.13.5
aiosignal                1.4.0
annotated-doc            0.0.4
annotated-types          0.7.0
anyio                    4.13.0
asttokens                3.0.1
async-timeout            5.0.1
attrs                    26.1.0
bitsandbytes             0.49.2
certifi                  2026.2.25
charset-normalizer       3.4.7
click                    8.3.2
colorama                 0.4.6
coloredlogs              15.0.1
contourpy                1.3.2
cycler                   0.12.1
datasets                 4.8.4
decorator                5.2.1
deepspeed                0.12.6
diffusers                0.37.1
dill                     0.4.1
einops                   0.6.1
einops-exts              0.0.4
exceptiongroup           1.3.1
executing                2.2.1
EzColorLog               1.0.3
fastapi                  0.136.0
filelock                 3.25.2
fonttools                4.62.1
frozenlist               1.8.0
fsspec                   2026.2.0
ftfy                     6.3.1
gitdb                    4.0.12
GitPython                3.1.46
h11                      0.14.0
hf-xet                   1.4.3
hjson                    3.1.0
httpcore                 0.17.3
httpx                    0.24.0
huggingface_hub          0.36.2
humanfriendly            10.0
idna                     3.11
importlib_metadata       9.0.0
ipython                  8.39.0
jedi                     0.19.2
jellyfish                1.2.1
Jinja2                   3.1.6
joblib                   1.5.3
kiwisolver               1.5.0
latex2mathml             3.81.0
llava                    1.2.2.post1  /storage/LVLM_Interpretation
markdown2                2.5.5
MarkupSafe               3.0.3
matplotlib               3.10.8
matplotlib-inline        0.2.1
mpmath                   1.3.0
multidict                6.7.1
multiprocess             0.70.19
networkx                 3.4.2
ninja                    1.13.0
numpy                    2.2.6
nvidia-cublas-cu12       12.4.5.8
nvidia-cuda-cupti-cu12   12.4.127
nvidia-cuda-nvrtc-cu12   12.4.127
nvidia-cuda-runtime-cu12 12.4.127
nvidia-cudnn-cu12        9.1.0.70
nvidia-cufft-cu12        11.2.1.3
nvidia-curand-cu12       10.3.5.147
nvidia-cusolver-cu12     11.6.1.9
nvidia-cusparse-cu12     12.3.1.170
nvidia-cusparselt-cu12   0.6.2
nvidia-ml-py             13.595.45
nvidia-nccl-cu12         2.21.5
nvidia-nvjitlink-cu12    12.4.127
nvidia-nvtx-cu12         12.4.127
open_clip_torch          3.3.0
opencv-python            4.13.0.92
packaging                26.0
pandas                   2.3.3
parso                    0.8.6
peft                     0.19.1
pexpect                  4.9.0
pillow                   12.2.0
pip                      26.0.1
platformdirs             4.9.6
prompt_toolkit           3.0.52
propcache                0.4.1
protobuf                 7.34.1
psutil                   7.2.2
ptyprocess               0.7.0
pure_eval                0.2.3
py-cpuinfo               9.0.0
pyarrow                  23.0.1
pydantic                 2.13.2
pydantic_core            2.46.2
Pygments                 2.20.0
pynvml                   13.0.1
pyparsing                3.3.2
python-dateutil          2.9.0.post0
pytz                     2026.1.post1
PyYAML                   6.0.3
regex                    2026.4.4
requests                 2.33.1
safetensors              0.7.0
scikit-learn             1.2.2
scipy                    1.15.3
segtok                   1.5.11
sentencepiece            0.1.99
sentry-sdk               2.58.0
setuptools               82.0.1
shortuuid                1.0.13
six                      1.17.0
smmap                    5.0.3
sniffio                  1.3.1
stack-data               0.6.3
starlette                1.0.0
svgwrite                 1.4.3
sympy                    1.13.1
tabulate                 0.10.0
threadpoolctl            3.6.0
timm                     0.9.16
tokenizers               0.15.1
torch                    2.6.0+cu124
torchaudio               2.6.0+cu124
torchvision              0.21.0+cu124
tqdm                     4.67.3
traitlets                5.14.3
transformers             4.37.2
triton                   3.2.0
typing_extensions        4.15.0
typing-inspection        0.4.2
tzdata                   2026.1
urllib3                  2.6.3
uvicorn                  0.44.0
wandb                    0.26.0
wavedrom                 2.0.3.post3
wcwidth                  0.6.0
wheel                    0.46.3
xxhash                   3.6.0
yake                     0.7.3
yarl                     1.23.0
zipp                     3.23.1

**Comment by srbhcs** [comment]
@szhu-bytedance @RuoyuChen10

Can you please help?

**Comment by RuoyuChen10** [comment]
> [@szhu-bytedance](https://github.com/szhu-bytedance) [@RuoyuChen10](https://github.com/RuoyuChen10)
> 
> Can you please help?

If you want to visualize MLLM, you can refer to this repository: https://github.com/RuoyuChen10/EAGLE

**Comment by srbhcs** [comment]
Hi @RuoyuChen10, thank you so much will look into it. But can someone please help me resolve the issue, as I am very interested in the iGOS++ based approach and really appreciate your work and want to utilize it. Any help and guidance I will deeply appreciate and acknowledge.

@RuoyuChen10 @szhu-bytedance

**Comment by srbhcs** [comment]
@xiaoying-xing

**Comment by xiaoying-xing** [comment]
Hi, thanks for your interest in this work. We have not tried llava-v1.6-mistral-7b yet. For different model structures, you may need to adjust the hyper-parameters accordingly. From the log you pasted, the code looks working correctly. However, the insertion and deletion scores for this specific sample are abnormal, which means the optimization didn't work well. 

I would suggest not focusing too much on a single specific sample, because not every sample is explainable. Our method is built on a prerequisite: the model’s output for that image must actually be related to the image. Otherwise, how can we visualize the relevant regions? From your logs, you can also see that the average deletion and insertion scores are fairly good, so overall the method appears to be working successfully.
